torch/csrc/jit/profiling_record.cpp

```diff
@@ -0,0 +1,79 @@
+#include <torch/csrc/jit/profiling_record.h>
+
+namespace torch {
+namespace jit {
+
+ProfilingRecord::ProfilingRecord(const std::shared_ptr<Graph>& g)
+    : profiled_graph_(g) {}
+
+void ProfilingRecord::instrumentBlock(Block* block) {
+  auto graph = block->owningGraph();
+
+  // iterating backwards allows us to easily insert profile nodes
+  // without affecting an iterator
+  for (auto it = block->nodes().rend(); it != block->nodes().rbegin(); --it) {
+    auto n = *it;
+    for (auto o : n->outputs()) {
+      if (!o->type()->isSubclass(TypeKind::TensorType)) {
+        continue;
+      }
+      auto pn = graph->create(prim::profile, 0);
```

Comment:
nit: I feel like the way this is organized into functions is weird. I would have expected:

```
Node* ProfiledTensorType::createCallback(std::function<void(Stack&)> callback, at::ArrayRef<Value> inputs);
```
Rather than addCallback. That way the details of a prim::profile node are contained in one function.